### ISSUE TITLE
vinyl: keep a stack of UPSERTS in vy_read_iterator

### ISSUE THREAD

**Issue by rtsisyk** [tarantool/tarantool#1833]
Instead of grouping several UPSERTS into a huge one and applying it to terminal (non-UPSERT) statement, it is better to find the terminal statement and then iterate back over UPSERTs applying them one by one.
Of course for this approach one have to implement something like **_prev_lsn_** method in all the iterators.

**Comment by alyapunov** [comment]
That will also fix a problem:
```
s = box.schema.space.create('test', {engine = 'vinyl'})
i = s:create_index('test')

s:replace{1, 0}
s:replace{2, 0}
box.snapshot()
s:upsert({1}, {{'+', 2, 1}})
s:upsert({2}, {{'+', 2, 1}})

tasks_done = 0
roadmap = {}

function f1()
    table.insert(roadmap, "1 start")
    box.begin()
    local t = s:get{1}
    s:replace{2, t[2]}
    box.commit()
    tasks_done = tasks_done + 1
    table.insert(roadmap, "1 end")
end

function f2()
    table.insert(roadmap, "2 start")
    box.begin()
    s:replace{1, 15}
    s:replace{2, 15}
    box.commit()
    tasks_done = tasks_done + 1
    table.insert(roadmap, "2 end")
end

fb1 = fiber.create(f1)
fb2 = fiber.create(f2)

while tasks_done ~= 2 do fiber.sleep(0.01) end

print(s:get{1}[2])
print(s:get{2}[2])
```
```
15
1
```

The invariant s:get{1}[2] == s:get{2}[2] is broken

**Comment by rtsisyk** [comment]
We discussed with @alyapunov that the background squasher with this optimization is more efficient than the foreground squasher in vy_range_set(). Probably it makes sense to remove foreground squasher completely.

**Comment by rtsisyk** [comment]
Blocked by #2090

**Comment by rtsisyk** [comment]
Moved to 1.7.6 together with #2090

**Comment by locker** [comment]
Accidentally fixed by 1e1c1fdbeddb7677b8bbde318ca85eabff78ca3c
